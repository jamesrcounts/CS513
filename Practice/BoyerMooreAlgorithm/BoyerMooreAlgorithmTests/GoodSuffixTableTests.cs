using ApprovalTests;
using ApprovalTests.Reporters;
using BoyerMooreAlgorithm;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace BoyerMooreAlgorithmTests
{
    [TestClass]
    [UseReporter(typeof(DiffReporter))]
    public class GoodSuffixTableTests
    {
        [TestMethod]
        public void B00001()
        {
            GoodSuffixTableTest("00001");
        }

        [TestMethod]
        public void B01010()
        {
            GoodSuffixTableTest("01010");
        }

        [TestMethod]
        public void B10000()
        {
            GoodSuffixTableTest("10000");
        }

        [TestMethod]
        public void BAOBAB()
        {
            GoodSuffixTableTest("BAOBAB");
        }

        [TestMethod]
        public void Case1()
        {
            GoodSuffixTableTest("ABABAB");
        }

        [TestMethod]
        [UseReporter(typeof(TortoiseDiffReporter))]
        public void Case2()
        {
            GoodSuffixTableTest("XBABAB");
        }

        [TestMethod]
        public void Case3()
        {
            GoodSuffixTableTest("ABCBAB");
        }

        private static void GoodSuffixTableTest(string pattern)
        {
            Approvals.VerifyAll(Program.GoodSuffixTable(pattern));
        }
    }
}