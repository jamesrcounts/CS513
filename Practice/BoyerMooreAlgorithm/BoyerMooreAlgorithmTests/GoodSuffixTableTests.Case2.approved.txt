1 => 4
2 => 6
3 => 2
4 => 6
5 => 6
